ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/EthSigner.java

```diff
@@ -92,27 +94,60 @@ public void run() {
     runner.start();
   }
 
-  private HttpClientOptions applyConfigTlsSettingsTo(final HttpClientOptions input) {
-    final HttpClientOptions result = new HttpClientOptions(input);
+  private WebClientOptions applyConfigTlsSettingsTo(final WebClientOptions input) {
+    final Optional<DownstreamTlsOptions> optionalDownstreamTlsOptions =
+        config.getDownstreamTlsOptions();
+    if (optionalDownstreamTlsOptions.isEmpty()
+        || !optionalDownstreamTlsOptions.get().isTlsEnabled()) {
+      return input;
+    }
+
+    final WebClientOptions result = new WebClientOptions(input);
+    final DownstreamTlsOptions downstreamTlsOptions = optionalDownstreamTlsOptions.get();
+    result.setSsl(true);
+
+    applyDownstreamTlsTrustOptions(
+        result, downstreamTlsOptions.getDownstreamTlsServerTrustOptions());
+    applyDownstreamClientAuthOptions(
+        result, downstreamTlsOptions.getDownstreamTlsClientAuthOptions());
 
-    result.setSsl(
-        config.getWeb3ProviderKnownServersFile().isPresent()
-            || config.getClientCertificateOptions().isPresent());
+    return result;
+  }
 
-    config
-        .getWeb3ProviderKnownServersFile()
-        .ifPresent(
-            knownServerFile -> result.setTrustOptions(whitelistServers(knownServerFile.toPath())));
+  private void applyDownstreamTlsTrustOptions(
+      final WebClientOptions result,
+      final Optional<DownstreamTrustOptions> optionalDownstreamTrustOptions) {
+
+    if (optionalDownstreamTrustOptions.isPresent()) {
+      final Optional<Path> optionalKnownServerFile =
+          optionalDownstreamTrustOptions.get().getKnownServerFile();
+      if (optionalKnownServerFile.isPresent()) {
+        final Path knownServerFile = optionalKnownServerFile.get();
+        try {
+          result.setTrustOptions(
+              whitelistServers(
+                  knownServerFile,
+                  optionalDownstreamTrustOptions.get().isCaSignedServerCertificateAllowed()));
+        } catch (RuntimeException e) {
+          throw new InitializationException("Failed to load known server file.", e);
+        }
+      } else if (!optionalDownstreamTrustOptions.get().isCaSignedServerCertificateAllowed()) {
+        throw new InitializationException(
+            "Downstream TLS CA-signed option cannot be disabled when known-server file is not specified");
+      }
+    }
+  }
 
-    if (config.getClientCertificateOptions().isPresent()) {
+  private void applyDownstreamClientAuthOptions(
```

Comment:
nit: Rename this more generically; this isn't explcitily a "downstream" thing - it's applying a Pkcs file to a webclient config, having the word "downstream" implies usage, rather than functionality (which means that the 2nd parameter should also be renamed to simply keystore?)

Reply:
method has been renamed.